### PR TITLE
added ssh key file option to connect to hosts with

### PR DESCRIPTION
**Pull request by willejs**
Add the ability to use ssh keys to auth with hosts.

### CODE REVIEW

**Comment by nilstoedtmann** [comment]
See also https://github.com/DemandLogic/cassandra_snapshotter/commit/da96b0f34775dccc752faf11394705bc074624f5

**Comment by willejs** [comment]
@nilstoedtmann I don't mind which one gets merged!

File: cassandra_snapshotter/main.py [hunk 1]
```diff
@@ -19,6 +19,9 @@ def run_backup(args):
     if args.password:
         env.password = args.password
 
+    if args.sshkey-filename:
```
Comment (tbarbugli) [inline cassandra_snapshotter/main.py]: syntax error here, perhaps you meant sshkey_filename?

**Comment by tbarbugli** [comment]
This seems like a great addition. Perhaps a super solution to this specific problem is to have a mechanism that allows us to send optional args straight to fabric. (eg. send every arg that starts with --fabric- to fabric option parser to configure it). what do you guys think about it?

**Comment by nilstoedtmann** [comment]
i dont mind either

@tbarbugli Have a look at other options we added, maybe some of it is interesting to others as well: https://github.com/DemandLogic/cassandra_snapshotter/commits/master

**Comment by nilstoedtmann** [comment]
+1 to the generic "--fabric-*" idea

**Comment by willejs** [comment]
@tbarbugli seems like a better idea really. could just loop through the args, and match on 'fabric-.*', then strip it and have a 1:1 mapping of args to options, then call getattr on them with the value as the argument?

**Comment by tbarbugli** [comment]
yes, i was thinking about something like that. It would be great to send
the list of options to fabric option parser so that we don't do to touch
fab's env ourselves.

**Comment by willejs** [comment]
@tbarbugli I hadnt read the code properly when making my previous assumptions. When you use the API it imports the env dictionary with populated with defaults it [seems](https://github.com/fabric/fabric/blob/master/fabric/api.py#L15). You cant set the options anywhere else, apart from with settings, which isnt appropriate.
Therefore looping through the dictionary and checking existence of the keys before setting and warning if they don't exist is perhaps better?

**Comment by ajhoffma** [comment]
is this pull request still active? Or @nilstoedtmann do you plan on pull-requesting your changes?

**Comment by tbarbugli** [comment]
added support for this with a newer PR, sorry for the long wait!

**Pull request closed without merging**